crates/sargon/src/system/clients/client/factor_instances_cache_client.rs

```diff
@@ -0,0 +1,361 @@
+use std::borrow::Borrow;
+
+use crate::prelude::*;
+
+#[derive(Debug, Clone)]
+pub struct FactorInstancesCacheClient {
+    file_system_client: Arc<FileSystemClient>,
+}
+
+impl FactorInstancesCacheClient {
```

Comment:
SELF: Check if we can remove ALL methods on this type and replace with `Deref` and `DerefMut`! Maybe using `derive_more` ?